florist/app/client_imports.tsx

```
@@ -12,12 +13,23 @@ function ClientImports(): null {
     return null;
 }
 
-const fetcher = (...args: Parameters<typeof fetch>) =>
-    fetch(...args).then((res) => {
+const fetcher = (url: string) => {
+    // Adding the authentication token to the request headers
+    const params: RequestInit = {
+        headers: new Headers(getAuthHeaders()),
+    };
+
+    return fetch(url, params).then((res) => {
+        if (res.status === 401) {
+            // redirecting to the login page in case any calls return a 401 (unauthorized) error
+            window.location.href = "/login";
+        }
         if (res.status != 200) {
             throw new Error(res.status.toString(), { cause: res.json() });
```

I'm not good with front end. So this will likely be a dumb question, but is this if statement still going to throw (even if the one above executes), and, if so, is that a problem?

---

Yes, and that's ok. For all calls that return a non-200 code we want to throw an exception so we can see the trace in the browser's console, but for 400 codes in specific we also want to redirect to the login page. The redirect has to happen before the throw otherwise it will be unreachable.

---

I see...the error isn't going to blow up the process, just log in the console? I was just worried we'd like kill the whole client.

---

There is a `catch` on the caller of this function that will handle this gracefully, but even if it wasn't caught it wouldn't stop the client.